src/test/tree/StrategyManangerUnit.tree

```diff
@@ -0,0 +1,107 @@
+├── StrategyManagerUnit.t.sol
+├── initialize
+│   └── given that initialize is called again
+│       └── it should revert
+├── depositIntoStrategy()
+│   ├── given that deposits paused
+│   │   └── it should revert
+│   ├── given the function is re-entered
+│   │   └── it should revert
+│   ├── given that the strategy is not whitelisted
+│   │   └── it should revert
+│   ├── given the token safeTransferFrom() reverts
+│   │   └── it should revert
+│   └── given that token safeTransferFrom() succeeds
+│       ├── given the staker has existing shares in strategy (not first deposit)
+│       │   └── it should increase shares, nonce. while stakerStrategyListLength is unchanged
+│       ├── given the staker has no existing shares in strategy (first deposit)
+│       │   └── stakerStrategyListLength increases by 1 and shares, nonce increase
+│       ├── given the staker has delegated to a operator ***
+│       │   └── it should deposit successfully with shares and nonce increase, including delegated shares
```

Comment:
probably want to add the same note as in the EigenPodManager tree
`(*** denotes that integration tests are needed to validate the path)`